### DISCUSSION TITLE
FileUpload fetch invalid URL image from external URL

### DISCUSSION THREAD

**Discussion by adityahilman** [filamentphp/filament#14072] [Help] [answered]
### Package

Form builder

### Package Version

v3.2

### How can we help you?

I am encountering an issue with the FileUpload component in Laravel Filament when trying to fetch images from an external URL, specifically from imagekit.io.
No issue when storing the URL in the database.

**Issue:** 
When I try to fetch an image using an external URL like https://imagekit.io/user/image-xxxx.jpeg, the FileUpload component incorrectly processes the URL and returns an invalid URL in the format: http://localhost:8000/https://imagekit.io/user/image-xxxx.jpeg.

**Expected Behavior:** 
The FileUpload component should correctly handle and display the external URL without appending the localhost base URL.

**Comment by leandrocfe** [comment]
Are you using imagekit as a driver? 
Like this? https://github.com/phuclh/flysystem-imagekit

**Reply by adityahilman** [reply]
No, I use Guzzle to upload the image with the POST method. Currently, there is no issue on the List page, where the thumbnail fetches the correct URL from ImageKit. The issue only occurs on the Edit page, where it fetches an invalid URL

**Comment by adityahilman** [comment, marked as answer]
Problem solved. 

I created a custom disk in filesystems.php and set the URL to the ImageKit URL.

> 
        'imagekit' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'visibility' => 'public',
            'url' => env('IMAGEKIT_URL_ENDPOINT'),
        ],
